test/ruby/test_rubyoptions.rb

```diff
@@ -307,6 +307,16 @@ def test_rjit_version
     end
   end
 
+  def test_enabled_gc
+    omit unless /linux|darwin/ =~ RUBY_PLATFORM
+
+    if RbConfig::CONFIG['shared_gc_dir'].length > 0
+      assert_match(/\+GC/, RUBY_DESCRIPTION)
+    else
+      assert_match(/^((?!\+GC(\[\w+\])?).)*$/, RUBY_DESCRIPTION)
```

Comment:
Can we just do this here?

```suggestion
      assert_no_match(/\+GC/, RUBY_DESCRIPTION)
```